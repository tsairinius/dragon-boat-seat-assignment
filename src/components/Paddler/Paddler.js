import React from 'react';
import PropTypes from 'prop-types';
import defaultBackground from '../../assets/img/profile_default_img_new.svg'
import activeBackground from '../../assets/img/profile_default_img_new_hover.svg'

function Paddler(props) {

    const profileImg = props.paddlerProfile.isActive ? activeBackground : defaultBackground;

    const style = {
        width: '50px',
        height: '50px',
        margin: '1px',
        float: 'left',
        textAlign: 'center',
        backgroundImage: "url(" + profileImg + ")",
        backgroundSize: 'cover',
        backgroundPosition: '30% 100%'
    };

    const handlePaddlerClick = () => props.handlePaddlerClick(props.paddlerProfile.id)
    
    return (
        <div style={style} onClick={handlePaddlerClick}>
            {props.paddlerProfile.name}
        </div>
    )
};

Paddler.propTypes = {
    paddlerProfile: PropTypes.object.isRequired,
    handlePaddlerClick: PropTypes.func.isRequired
}


export default Paddler;