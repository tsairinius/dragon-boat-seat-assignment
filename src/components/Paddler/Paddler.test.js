import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Paddler from './Paddler';

afterEach(cleanup);

it('renders Paddler component without crashing', () => {
  const container = document.createElement('div');
  ReactDOM.render(<Paddler />, container);
});

it('rendered Paddler matches snapshot', () => {
    const tree = renderer.create(<Paddler />).toJSON();
    expect(tree).toMatchSnapshot();
});
