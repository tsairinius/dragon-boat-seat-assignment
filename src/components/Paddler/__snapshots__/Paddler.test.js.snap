// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered Paddler matches snapshot 1`] = `
<div
  style={
    Object {
      "border": "1px solid green",
    }
  }
/>
`;
